### PR TITLE
New package: JamesDSP-2.7.0

### PR DESCRIPTION
**Pull request by Anachron**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):

### CODE REVIEW

**Comment by Anachron** [comment]
ToDo: 
- Create headless setup

File: srcpkgs/JamesDSP/template [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ ${_liquidequalizerwidget_commit}.tar.gz
+ ${_liveprogide_commit}.tar.gz"
+
+case "${XBPS_TARGET_MACHINE}" in
+ *-musl) broken="subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/jdspController.c:66:24: error: storage size of 'tv' isn't known";;
+ aarch64) broken="https://github.com/Audio4Linux/JDSP4Linux/issues/187";;
+esac
+
+post_extract() {
+	find "src/subprojects" -type d -empty -delete
```
Comment (tranzystorekk) [inline srcpkgs/JamesDSP/template:38]: Doesn't seem necessary

**Comment by Anachron** [comment]
Difference between `JamesDSP`:

```
[18.08.2024-22:15:37] (JamesDSP) xins:start
Do you want to continue? [Y/n] n
14 packages will be installed:

  double-conversion-3.3.0_1 
  libb2-0.98.1_1 
  qt6-core-6.7.2_2 
  qt6-dbus-6.7.2_2 
  libmd4c-0.4.8_1 
  xcb-util-cursor-0.1.4_1 
  xcb-util-keysyms-0.4.1_1 
  tslib-1.23_1 
  libproxy-0.4.18_3 
  qt6-network-6.7.2_2 
  qt6-gui-6.7.2_2 
  qt6-widgets-6.7.2_2 
  qt6-svg-6.7.2_1 
  JamesDSP-2.7.0_1 

Size required on disk:          45MB
Space available on disk:        15GB

Aborting!
[18.08.2024-22:15:38] (JamesDSP) xins:finish (RC=0)
```

And `JamesDSP-headless`:

```
[18.08.2024-22:14:23] (JamesDSP-headless) xins:start
Do you want to continue? [Y/n] n
5 packages will be installed:

  double-conversion-3.3.0_1 
  libb2-0.98.1_1 
  qt6-core-6.7.2_2 
  qt6-dbus-6.7.2_2 
  JamesDSP-headless-2.7.0_1 

Size required on disk:          19MB
Space available on disk:        15GB

Aborting!
[18.08.2024-22:14:24] (JamesDSP-headless) xins:finish (RC=0)
```

**Comment by Anachron** [comment]
Not sure if I should add two seperate commits for the packages, but I've added it as one as `JamesDSP-headless` is practically the same package just with different built options.

**Comment by Anachron** [comment]
This is ready for review now.

**Comment by tranzystorekk** [comment]
To me, splitting the package when the GUI variant offers all the functionality that the headless one does, doesn't seem justified just to avoid ~20MB of dependencies

**Comment by Anachron** [comment]
It's not even half the size and dependencies shrinks from 14 to 5. How is that not worth it?

**Comment by tranzystorekk** [comment]
The templates are redundantly duplicated, conflict each other. I don't see a precedent in existing packages for maintaining an additional burden like this.

**Comment by ahesford** [comment]
Using two templates that produce conflicting packages is not good. I recommend either
- Just pick one, possibly adding a build option to allow others to build the other variant (although build options generally suck anyway); or
- Build both variants at one, and give the headless one a name that won't conflict with the GUI variant.

**Comment by Anachron** [comment]
They conflict only because they use a shared library (libjamesdsp).

I will probably go the build option way as I don't feel like arguing against this pushback. 

Now I gotta figure out how to check build options in IF-statements to conditionally set distfiles and vextract them.

To be updated.

**Comment by ahesford** [comment]
I didn't realize that the distfiles were different. This is going to make for a convoluted template, so I recommend you pick whichever variant you want to use regularly and package only that.

**Pull request closed without merging**